mw/xml_dump/iteration/iterator.py

```diff
@@ -140,6 +143,9 @@ def load_site_info(cls, element):
         namespaces = {}
 
         for sub_element in element:
+
+            if sub_element.tag == 'siteinfo':
+                return cls.load_site_info(sub_element)
```

Comment:
OK i see now why this looks weird. I'll take another look. It's been a while since I did this so I'll see if it's a mistake or just something strange going on with Wikia dumps.